openwisp_notifications/api/views.py

```diff
@@ -198,11 +206,64 @@ def perform_create(self, serializer):
         )
 
 
+class OrganizationNotificationSettingView(GenericAPIView):
+    permission_classes = [IsAuthenticated, PreferencesPermission]
+    serializer_class = NotificationSettingUpdateSerializer
+
+    def post(self, request, user_id, organization_id):
+        notification_settings = NotificationSetting.objects.filter(
+            organization_id=organization_id, user_id=user_id
+        )
+        serializer = self.get_serializer(data=request.data)
+        if serializer.is_valid():
+            for notification_setting in notification_settings:
+                serializer.update(notification_setting, serializer.validated_data)
+            return Response(status=status.HTTP_200_OK)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+class NotificationPreferenceView(GenericAPIView):
+    permission_classes = [IsAuthenticated, PreferencesPermission]
+    serializer_class = NotificationSettingUpdateSerializer
+
+    def get(self, request, user_id):
+        notification_settings, created = NotificationSetting.objects.get_or_create(
+            user_id=user_id,
+            organization=None,
+            type=None,
+            defaults={'email': True, 'web': True},
+        )
+        serializer = self.get_serializer(notification_settings)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    def post(self, request, user_id):
+        serializer = self.get_serializer(data=request.data)
```

Comment:
Can we use the `NotificationSettingSerializer`  also for updating the global notifications?

Reply:
You are right, updated.